### ISSUE TITLE
M600 - Wait on Temp Change Issue

### ISSUE THREAD

**Issue by zjed42** [andrivet/ADVi3pp#114]
**- Description:** If M600 is issued immediately after a Extruder temp change, the the hot end pauses at current z-height while temp change occurs
**- Expected behavior:** Once the M600 command is issued, hot end should move to park position and begin filament unload, regardless of the current or set hot end temp. at minimum, a small Z-up should occur to eliminate risk of melting part.
**- Actual behavior:** M600 issued, then hot end pauses at current z-height. "Waiting on extruder temp" (or something similar) Message is displayed. once desired temp is reached, the M600 routine is resumed. 
**- Steps to reproduce:**
  - Start print at some temp, then change to something lower, or higher.
  - Send M600
  - Observe the hot end pause while temp change occurs. 

**Justification:** I was swapping between two filaments with different temps, M600 routine doesn't seem to allow temp change during filament swap, so I tried issuing a temp change just prior, then sent M600, That's when I noticed that the hot end paused and melted a spot on the print.

**Comment by andrivet** [comment]
This is a very specific scenario and, as far as I understand, it is not a bug but you are requesting a change of behavior. For me, this issue overlap with your other issue #115.

**Comment by zjed42** [comment]
Agreed that it's specific. I mainly summited as a bug due to the risk of print failure. 

Thanks for all of your efforts!

**Comment by andrivet** [comment]
So I am closing this in favor of #115
